packages/master/master.js

```diff
@@ -2006,6 +2029,15 @@ async function startServer(args) {
 	await fs.ensureDir(masterConfig.get("master.database_directory"));
 
 	db.slaves = await loadMap(masterConfig.get("master.database_directory"), "slaves.json");
+	// live updating
+	db.slaves.updateGui = function(key) {
+		for (let control of controlConnections) {
+			if (control.doLiveUpdateSlaves) {
+				control.updateSlaves(key);
+			}
+		}
+	};
```

Comment:
Once again you're modifying an instance of a built-in Map class. The type of `db.slaves` is `Map`, not `MapWhichMayHaveAnUpdateGuiMethod`. This should be a stand alone function.

Also this is inefficient. You do the work of retrieving the slave record and constructing the update for every control connection instead of doing the work once and sending the data to each control connection. The code in ControlConnection.updateSlaves could be moved outside the for loop here and then the message with the computed item sent to each connection inside the loop.